polar-core/src/vm.rs

```diff
@@ -867,12 +861,12 @@ impl PolarVirtualMachine {
             .query_start_time
```

Comment:
Might consider breaking the current query duration calculation out into a separate function. Something like:

```rs
    #[cfg(not(target_arch = "wasm32"))]
    fn query_duration(&self) -> u64 {
        let now = std::time::Instant::now();
        let start = self.query_start_time.expect("Query start not recorded");
        (now - start).as_millis() as u64
    }

    #[cfg(target_arch = "wasm32")]
    fn query_duration(&self) -> u64 {
        let now: f64 = js_sys::Date::now();
        let start = self.query_start_time.expect("Query start not recorded");
        (now - start) as u64
    }

    // TODO (dhatch): How do we reliably not do this when debugging.
    fn check_timeout(&self) -> PolarResult<()> {
        let elapsed = self.query_duration();
        if elapsed > self.query_timeout_ms {
            return Err(error::RuntimeError::QueryTimeout {
                msg: format!(
                    "Query running for {}ms. Exceeded query timeout of {}ms",
                    elapsed, self.query_timeout_ms
                ),
            }
            .into());
        }
        Ok(())
    }
```